modules/aws/ec2.go

```diff
@@ -63,6 +63,63 @@ func GetPrivateIpsOfEc2InstancesE(t *testing.T, instanceIDs []string, awsRegion
 	return ips, nil
 }
 
+// GetPrivateHostnameOfEc2Instance gets the private IP address of the given EC2 Instance in the given region.
+func GetPrivateHostnameOfEc2Instance(t *testing.T, instanceID string, awsRegion string) string {
+	ip, err := GetPrivateHostnameOfEc2InstanceE(t, instanceID, awsRegion)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return ip
+}
+
+// GetPrivateHostnameOfEc2InstanceE gets the private IP address of the given EC2 Instance in the given region.
+func GetPrivateHostnameOfEc2InstanceE(t *testing.T, instanceID string, awsRegion string) (string, error) {
+	hostnames, err := GetPrivateHostnamesOfEc2InstancesE(t, []string{instanceID}, awsRegion)
+	if err != nil {
+		return "", err
+	}
+
+	hostname, containsHostname := hostnames[instanceID]
+
+	if !containsHostname {
+		return "", HostnameForEc2InstanceNotFound{InstanceId: instanceID, AwsRegion: awsRegion, Type: "private"}
+	}
+
+	return hostname, nil
+}
+
+// GetPrivateHostnamesOfEc2Instances gets the private IP address of the given EC2 Instance in the given region. Returns a map of instance ID to IP address.
+func GetPrivateHostnamesOfEc2Instances(t *testing.T, instanceIDs []string, awsRegion string) map[string]string {
+	ips, err := GetPrivateHostnamesOfEc2InstancesE(t, instanceIDs, awsRegion)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return ips
+}
+
+// GetPrivateHostnamesOfEc2InstancesE gets the private IP address of the given EC2 Instance in the given region. Returns a map of instance ID to IP address.
+func GetPrivateHostnamesOfEc2InstancesE(t *testing.T, instanceIDs []string, awsRegion string) (map[string]string, error) {
+	ec2Client, err := NewEc2ClientE(t, awsRegion)
+	if err != nil {
+		return nil, err
+	}
+	input := ec2.DescribeInstancesInput{InstanceIds: aws.StringSlice(instanceIDs)}
+	output, err := ec2Client.DescribeInstances(&input)
```

Comment:
Is this a paginated API?

Reply:
Yes, but this is actually a copy paste job so all the similar methods (get private/public ip) have this problem. I'll add a global comment to mention that we need pagination here, although I suspect that most people using terratest won't hit the limit ([1000](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#DescribeInstancesInput)).